### PR TITLE
Use the new Cosmos SDK

### LINKED ISSUE #14123: CosmosDB provider => 400 errors
I've integrated Cosmos DB in my ASP.NET Core API project, using the latest EF Core package that ships with a Cosmos DB provider.

I'm able to successfully use this provider when running locally in my local ASF cluster, connecting to a local Cosmos DB emulator instance. However, when trying to either run in local ASF or Azure-hosted ASF, I'm unable to connect to the Azure Cosmos DB instance. For some reason I keep getting a 400 BadRequest when initializing the seed for the database/collection.

```
Exception message: BadRequest
Stack trace:
BadRequest
     at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosClient.CreateDocumentCollectionIfNotExistsOnceAsync(DbContext _, String collectionId, CancellationToken cancellationToken)
       at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.ExecuteImplementationAsync[TState,TResult](Func`4 operation, Func`4 verifySucceeded, TState state, CancellationToken cancellationToken)
       at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.ExecuteImplementationAsync[TState,TResult](Func`4 operation, Func`4 verifySucceeded, TState state, CancellationToken cancellationToken)
       at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosDatabaseCreator.EnsureCreatedAsync(CancellationToken cancellationToken)
       at Infrastructure.SystemDataContextSeed.SeedAsync(SystemDataContext context) in Infrastructure\SystemDataContextSeed.cs:line 21
       at xtensions.IWebHostExtensions.Seed(IWebHost webhost) in Extensions\IWebHostExtensions.cs:line 23
       at Api.<>c__DisplayClass1_0.<CreateServiceInstanceListeners>b__1(String url, AspNetCoreCommunicationListener listener) in AccountApi.cs:line 47
       at Microsoft.ServiceFabric.Services.Communication.AspNetCore.AspNetCoreCommunicationListener.OpenAsync(CancellationToken cancellationToken)
       at Microsoft.ServiceFabric.Services.Runtime.StatelessServiceInstanceAdapter.OpenCommunicationListenersAsync(CancellationToken cancellationToken)
```

Configuration:
"Cosmos": {
        "EndPoint": "https://xxxx-sqlapi.documents.azure.com:443",
        "AuthKey": "xxx==",
        "DatabaseName": "AccountService"
      },
Extension method to hook up the Cosmos DB provider:
```c#


    public static IServiceCollection AddCosmosSettings(this IServiceCollection services, IConfiguration configuration)
            {
                services.AddEntityFrameworkCosmos()
                       .AddDbContext<SystemDataContext>(options =>
                       {
                           options.UseCosmos(configuration["Cosmos:EndPoint"], configuration["Cosmos:AuthKey"], configuration["Cosmos:DatabaseName"]); ;
                       },
                           ServiceLifetime.Scoped  //Showing explicitly that the DbContext is shared across the HTTP request scope (graph of objects started in the HTTP request)
                       );
    
                return services;
            }
```
What I've tried:

- Analysing log analytics, shows no logging for these requests
- Removing the EnsureCreated and just directly insert data
- Create the collection manually (SystemDataContext)
- Create the database manually (AccountService)
- Removing networking firewall (Allow all networks) in azure

### Further technical details
EF Core version: 2.2.0
Database Provider: Microsoft.EntityFrameworkCore.Cosmos 2.2.0-preview3-35497
Operating system: Azure Service Fabric - Windows Server Datacenter Core
IDE: VS 2017 15.9.3

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Sync APIs will be delegated to async ones for now.

Fixes #14123

### CODE REVIEW

File: src/EFCore.Cosmos/EFCore.Cosmos.csproj [hunk 1]
```diff
@@ -22,6 +22,7 @@
   <ItemGroup>
     <PackageReference Include="StyleCop.Analyzers" Version="$(StyleCopAnalyzersPackageVersion)" PrivateAssets="All" />
     <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonPackageVersion)" />
```
Comment (smitpatel) [inline src/EFCore.Cosmos/EFCore.Cosmos.csproj]: Do we still need this reference?

File: src/EFCore.Cosmos/Storage/Internal/CosmosClientWrapper.cs [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+            CancellationToken cancellationToken = default)
+            => _executionStrategyFactory.Create().ExecuteAsync(
+                (object)null, DeleteDatabaseOnceAsync, null, cancellationToken);
+
+        public async Task<bool> DeleteDatabaseOnceAsync(
+            DbContext _,
+            object __,
+            CancellationToken cancellationToken = default)
+        {
+            var database = await GetDatabaseAsync(cancellationToken);
```
Comment (smitpatel) [inline src/EFCore.Cosmos/Storage/Internal/CosmosClientWrapper.cs]: If _database == null then GetDatabaseAsync will create the database and then we are deleting database in this method. It does not seem to be good behavior. Creating database just to delete it straight ahead.

Comment (AndriySvyryd) [reply src/EFCore.Cosmos/Storage/Internal/CosmosClientWrapper.cs]: I agree, but there doesn't seems to be a way around this in the new SDK. I've already sent this and other feedback to the SDK team.

File: src/EFCore.Cosmos/EFCore.Cosmos.csproj (hunk 1 above)
Comment (bricelam) [reply src/EFCore.Cosmos/EFCore.Cosmos.csproj]: Yes, `dotnet-ef` uses it.

Comment (AndriySvyryd) [reply src/EFCore.Cosmos/EFCore.Cosmos.csproj]: "Microsoft.Azure.Cosmos" references it already. But I think we should still keep this as we do use it directly and "Microsoft.Azure.Cosmos" might drop it in the future.

**Pull request merged**